stdlib/public/SwiftShims/swift/shims/SwiftStdint.h

```diff
@@ -24,7 +24,7 @@
 
 // Clang has been defining __INTxx_TYPE__ macros for a long time.
 // __UINTxx_TYPE__ are defined only since Clang 3.5.
-#if !defined(__APPLE__) && !defined(__linux__) && !defined(__OpenBSD__) && !defined(__wasi__)
+#if !defined(__APPLE__) && !defined(__linux__) && !defined(__OpenBSD__) && !defined(__wasi__) && __STDC_HOSTED__
```

Comment:
https://github.com/llvm/llvm-project/blob/main/clang/lib/Headers/stdint.h provides the necessary definitions, which should be available in freestanding mode as well. This feels like it might be a missing flag or an incorrect clang configuration.

Reply:
So on 20.04 this looks to be true:
```
# 1 "/home/build-user/ws/public/swift-next/build/buildbot_linux/llvm-linux-aarch64/lib/clang/17/include/stdint.h" 1 3
```

On both 20.04 and 22.04 we have:
```
symlinking the system headers (/usr/include/c++) into the local clang build directory (/home/build-user/build/buildbot_linux/llvm-linux-x86_64/include).
```

There is a `stdint.h` in `/usr/include/c++/9/tr1/stdint.h` on 20.04, but it appears to not be used. But on 22.04 we are now installing libc++ (https://github.com/swiftlang/swift-docker/commit/7b194efd9fa36a10f10d45c428a15edcc17fb95f and https://github.com/swiftlang/swift-docker/commit/d00efae876b9e54f90cd6b242dab353b0cc4f76e). We then pick up that `stdint.h` instead:
```
/home/build-user/build/buildbot_incremental/llvm-linux-x86_64/bin/../include/c++/v1/stdint.h
```

That contains the `#error` and 💥

Reply:
https://github.com/swiftlang/swift/pull/76761